### PR TITLE
Suggestions for answering stackoverflow question

### PR DESCRIPTION
**Pull request by ESloman**
So I think the `400 bad request` issue you're getting is easily solvable. Below I've made the changes to do that.

Effectively, if I send a message the bot will try to get a response from `responses.get_response`, and if it doesn't match any of the conditions it will always return `None`. You then don't do any checks so it's automatically sent; thus the error you're getting. My simple fix is to not do anything if `response` is None/False/empty.

I've also tidied up a bit.

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -1,66 +1,53 @@
...
 import responses
 
+# user id of the user we want to delete links for
+MY_ID = 189458414764687360
+
 async def send_message(message, user_message, is_private):
     try:
         response = responses.get_response(user_message)
-        await message.author.send(response) if is_private else await message.channel.send(response)
+        if not response:
```
Comment (ESloman) [inline bot.py:10]: This here is the main check for making sure we're not sending an empty message.

File: responses.py [hunk 2]
```diff
@@ -1,26 +1,17 @@
...
 
+    rock_paper_scissors = ["rock", "paper", "scissors"]
 
+    if p_message == "hello":
+        return "Hey there!"
 
+    if p_message == "roll":
+        return str(random.randint(1, 6))
+ 
+    if p_message in rock_paper_scissors:
```
Comment (ESloman) [inline responses.py:16]: If we make all our values in the `rock_paper_scissors` list lowercase strings, then we can simplify the 3 if statements into one.

**Review by warlordgrog — APPROVED** [review]
thank you very much but it gives me a 2023-02-17 18:40:29 ERROR    discord.client Ignoring exception in on_message
Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\pythonProject\venv\lib\site-packages\discord\client.py", line 441, in _run_event
    await coro(*args, **kwargs)
  File "C:\Users\Administrator\Desktop\pythonProject\bot.py", line 38, in on_message
    if message.author.id == MY_ID and message.startswith("https"):
AttributeError: 'Message' object has no attribute 'startswith' error when i send a gif also i am sorry if i am wasting your time whit these questions im quite new to programming.

**Comment by ESloman** [comment]
Ah I know exactly what that is. I'll sort that later when I've got some time

**Comment by ESloman** [comment]
I added the fix.

You were doing `message.startswith` directly, but `message` is still the `discord.Message` object. Should be `message.content.startswith` as that'll be a string.

**Comment by warlordgrog** [comment]
now it works thank you very much for your help

**Comment by ESloman** [comment]
No worries - I'll delete this PR and fork shortly so make sure that you've got it saved somewhere.

**Comment by ESloman** [comment]
I don't have write access to the repo, so if you wanted the changes merged you will have to do that yourself :)

**Pull request merged**